### ISSUE TITLE
Cannot rename ECR repository

### ISSUE THREAD

**Issue by aschleck** [pulumi/pulumi-aws-native#612]
### What happened?

I created a repository via `ecr.Repository(name)` and then realized I wanted a non-random name. But when I changed it to `ecr.Repository(name, repository_name=name)` this provider tried to rename the object instead of deleting and recreating it.

### Steps to reproduce

Create a repository with

```
from pulumi_aws_native import ecr
ecr.Repository("repo")
```

### Expected Behavior

I expected the object to be deleted and recreated

### Actual Behavior

```
Updating (prod):
     Type                          Name            Status                  Info
     pulumi:pulumi:Stack           platform-prod   **failed**              1 error
 ~   └─ aws-native:ecr:Repository  platform/repo   **updating failed**     [diff: ~repositoryName]; 1 error

Diagnostics:
  aws-native:ecr:Repository (platform/repo):
    error: operation error CloudControl: UpdateResource, https response error StatusCode: 400, RequestID: 1edbd29b-507f-448c-b118-1424bf8cae8f, NotUpdatableException: Invalid patch update: createOnlyProperties [/properties/RepositoryName] cannot be updated
```

### Output of `pulumi about`

```
CLI
Version      3.39.1
Go Version   go1.19
Go Compiler  gc

Plugins
NAME           VERSION
aws-native     0.24.0
google-native  0.25.0
python         unknown

Host
OS       ubuntu
Version  22.04
Arch     x86_64

This project is written in python: executable='<snip>' version='3.10.2
'

Backend
Name           <snip>
URL            <snip>
User           april
Organizations

Dependencies:
NAME                  VERSION
pip                   22.0.3
pulumi-aws-native     0.24.0
pulumi-google-native  0.25.0
setuptools            60.9.3

Pulumi locates its logs in /tmp by default
warning: Failed to get information about the current stack: No current stack
```

### Additional context

Pulumi is much cooler than I realized, thank you for working on this!

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by viveklak** [comment]
Seems the same problem as #584. Seems like we are not appropriately marking createOnlyProperties as replaceOnChange.

**Comment by mjeffryes** [comment]
This should be resolved by #1053; changes to repository name will result in a replace rather than an update.
